src/Core/Validator/AvatarValidator.php

```diff
@@ -19,4 +20,14 @@ class AvatarValidator extends AbstractValidator
             'max:2048'
         ]
     ];
+
+    public function assertValid(array $attributes)
+    {
+        $validator = $this->makeValidator($attributes);
+
+        if ($validator->fails()) {
+            unlink($attributes['avatar']->avatarPath);
```

Comment:
You should be able to use `$attributes['avatar']->getRealPath()` here: http://php.net/manual/en/splfileinfo.getrealpath.php